frameworks/PHP/imi/ApiServer/Controller/IndexController.php

```diff
@@ -117,4 +218,63 @@ public function update($queries)
         return $list;
     }
 
+    /**
+     * @Action
+     *
+     * @return void
+     */
+    public function updateQueryBuilder($queries)
+    {
+        if($queries > 1)
+        {
+            $queryCount = min($queries, 500);
+        }
+        else
+        {
+            $queryCount = 1;
+        }
+        $list = [];
+        while ($queryCount--)
+        {
+            $id = mt_rand(1, 10000);
+            Db::query()->from('World')->where('id', '=', $id)->update([
+                'randomNumber'  =>  mt_rand(1, 10000),
+            ]);
+            $list[] = Db::query()->from('World')->field('id', 'randomNumber')->where('id', '=', $id)->select()->get();
+        }
+        return $list;
+    }
+
+    /**
+     * @Action
+     *
+     * @return void
+     */
+    public function updateRaw($queries)
+    {
+        if($queries > 1)
+        {
+            $queryCount = min($queries, 500);
+        }
+        else
+        {
+            $queryCount = 1;
+        }
+        $list = [];
+        $db = Db::getInstance();
+        $stmtSelect = $db->prepare('SELECT id, randomNumber FROM World WHERE id = ?');
+        $stmtUpdate = $db->prepare('UPDATE World SET randomNumber = :randomNumber WHERE id = :id');
+        while ($queryCount--)
+        {
+            $id = mt_rand(1, 10000);
+            $stmtUpdate->execute([

```

Comment:
This isn't following the rules of the update test: https://github.com/TechEmpower/FrameworkBenchmarks/wiki/Project-Information-Framework-Tests-Overview#database-updates

First, perform a select, map it to an object, update the `randomNumber` field on that object, then persist it to the database.

Reply:
> This isn't following the rules of the update test: https://github.com/TechEmpower/FrameworkBenchmarks/wiki/Project-Information-Framework-Tests-Overview#database-updates
> 
> First, perform a select, map it to an object, update the `randomNumber` field on that object, then persist it to the database.

I modified the code, please review again